DataOnion/RedisManager.cs

```diff
@@ -0,0 +1,37 @@
+using StackExchange.Redis;
+
+namespace DataOnion
+{
+    public interface IRedisManager
+    {
+        IDatabase GetDatabase();
+    }
+
+    public class RedisManager : IRedisManager
+    {
+        private readonly string _connectionString;
+        private ConnectionMultiplexer? _existingConnection = null;
+
+        private ConnectionMultiplexer _connection
+        {
+            get
+            {
+                if (_existingConnection == null || !_existingConnection.IsConnected)
+                {
+                    _existingConnection = ConnectionMultiplexer.Connect(_connectionString);
+                }
+
+                return _existingConnection;
+            }
+        }
```

Comment:
we should add logger lines for connecting to redis and any errors that occur doing so